cerberus/aws_auth.py

```diff
@@ -22,18 +22,19 @@ class AWSAuth(object):
     """Class to authenticate with an IAM Role"""
     HEADERS = {"Content-Type": "application/json", "X-Cerberus-Client": "CerberusPythonClient/" + CLIENT_VERSION}
 
-    def __init__(self, cerberus_url, region):
+    def __init__(self, cerberus_url, region, aws_creds=None):
```

Comment:
If this is going to be on the constructor, I think it should be a Session, so that when creds expire, the Session can try to get creds again.

Reply:
There's no way to refresh the token though. Also the Cerberus token is eagerly fetched. So adding a set method will require the token to be lazily fetched.